src/mmrelay/async_db_pool.py

```
@@ -0,0 +1,398 @@
+    """Cleanup async connection pools on application shutdown."""
+    await close_all_async_pools()
+
+
+# Register cleanup function to be called when the module is garbage collected
+def _cleanup_atexit():
+    """Cleanup function for atexit that handles async cleanup safely."""
+    try:
+        loop = asyncio.get_event_loop()
+        if loop.is_running():
+            # If loop is running, create a task
+            asyncio.create_task(async_cleanup())
```

Ruff (0.14.3)

Store a reference to the return value of `asyncio.create_task`

(RUF006)